src/dune_rules/pkg_rules.ml

```diff
@@ -264,6 +265,7 @@ module Pkg = struct
     ; build_command : Build_command.t option
     ; install_command : Dune_lang.Action.t option
     ; depends : t list
+    ; toolchain_depends : Dune_pkg.Toolchain.Version.t list
```

Comment:
Can you explain why this is a list? Naïvely I would expect this to be one single compiler that was picked.

Reply:
It's a list to represent the rare but possible situation where a package depends on both `ocaml-base-compiler` and `ocaml-variants`. However I think elsewhere in the code we make this case an error, so I'll change it to be an option instead for simplicity.